algorithms/refinement/reflection_manager.py

```diff
@@ -185,6 +185,19 @@ def __init__(self, reflections,
     # keep track of the original indices of the reflections
     reflections['iobs'] = flex.size_t_range(len(reflections))
 
+    #Check for monotonically increasing value range. If not, ref_table isn't sorted,
+    # and proceed to sort by id and panel. This is required for the C++ extension
+    # modules to allow for nlogn subselection of values used in refinement.
+    l_id = reflections["id"]
+    id0 = l_id[0]
+    for ii in xrange(1,len(l_id)):
```

Comment:
I was surprised how fast this Python loop is. I tested with a flex.size_t with 1 million elements (not inconceivable for a large synchrotron MX dataset) and this completes in about 0.8s on my desktop